### PR TITLE
Add extender/ScanMonitor script

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

File: other/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,9 @@ All notable changes to the 'other' section of this repository will be documented
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+### 2024-08-30
+- Added af-plans/ApiScanExample.yaml
```
Comment (psiinon) [inline other/CHANGELOG.md:7]: Looks like I should have included this in a prev PR :/

File: extender/ScanMonitor.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+var URI = Java.type("org.apache.commons.httpclient.URI");
+
+var extAscan = control.getExtensionLoader().getExtension("ExtensionActiveScan");
+var inMemoryStats = control
+  .getExtensionLoader()
+  .getExtension("ExtensionStats")
+  .getInMemoryStats();
+
+var timer = new Timer();
+var timerSecs = 10 * 1000; // Check every 10 secs
```
Comment (kingthorin) [inline extender/ScanMonitor.js:21]: Use TimeUnit?

File: extender/ScanMonitor.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    site == null
+      ? inMemoryStats.getStat(stat)
+      : inMemoryStats.getStat(site, stat);
+  return val == null ? 0 : val;
+}
+
+// Response times are recorded in logarithmic time slices
+function getLongRespStats(site) {
+  return (
+    getStat(site, "stats.responseTime.16384") +
```
Comment (kingthorin) [inline extender/ScanMonitor.js:50]: Are these ex: 16,32,64 sec? If so worth commenting?

File: extender/ScanMonitor.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    var connFails = getStat(null, "stats.network.send.failure");
+    // All HTTP response codes are recorded, so you can add checks for 401, 403 etc etc
+    var stats401 = getStat(site, "stats.code.401");
+    var stats500 = getStat(site, "stats.code.500");
+    // Auth fails are only relevant for authenticated scans
+    var authFails = getStat(site, "stats.auth.failure");
+    var longResp = getLongRespStats(site);
+
+    if (log) {
+      print("  401 resps:\t" + stats401);
```
Comment (kingthorin) [inline extender/ScanMonitor.js:73]: 403 too?

**Comment by kingthorin** [comment]
Looks like it needs a spotlessApply

File: extender/ScanMonitor.js (hunk 2 above)
Comment (psiinon) [reply extender/ScanMonitor.js:21]: Too much of a pain in a script? I want this to be easy for people to change..

File: extender/ScanMonitor.js (hunk 3 above)
Comment (psiinon) [reply extender/ScanMonitor.js:49]: `Response times are recorded in logarithmic time slices` above :)

File: extender/ScanMonitor.js (hunk 4 above)
Comment (psiinon) [reply extender/ScanMonitor.js:72]: We dont fail on these, so they are just examples..

**Comment by psiinon** [comment]
Had to exclude from the spotless check, after formatting it though :)

File: extender/ScanMonitor.js (hunk 3 above)
Comment (kingthorin) [reply extender/ScanMonitor.js:49]: Clarify that it's ms then?

Comment (psiinon) [reply extender/ScanMonitor.js:49]: Done

**Review by kingthorin — APPROVED** [review]

File: other/CHANGELOG.md (hunk 1 above)
Comment (thc202) [reply other/CHANGELOG.md:7]: I thought it was :)

**Comment by thc202** [comment]
Thank you!

**Pull request merged**